src/storage/mvcc/txn.rs

```diff
@@ -1311,26 +1317,125 @@ mod tests {
         must_commit_err(&engine, k, 24, 25);
         must_rollback(&engine, k, 24);
 
+        // Acquire lock on a prewritten key should fail.
+        must_acquire_pessimistic_lock(&engine, k, k, 26, 26);
+        must_pessimistic_locked(&engine, k, 26, 26);
+        must_pessimistic_prewrite_delete(&engine, k, k, 26, 26, true);
+        must_locked(&engine, k, 26);
+        must_acquire_pessimistic_lock_err(&engine, k, k, 26, 26);
+        must_locked(&engine, k, 26);
+
+        // Acquire lock on a committed key should fail.
+        must_commit(&engine, k, 26, 27);
+        must_unlocked(&engine, k);
+        must_get_none(&engine, k, 28);
+        must_acquire_pessimistic_lock_err(&engine, k, k, 26, 26);
+        must_unlocked(&engine, k);
+        must_get_none(&engine, k, 28);
+        // Pessimistic prewrite on a committed key should fail.
+        must_pessimistic_prewrite_put_err(&engine, k, v, k, 26, 26, true);
+        must_unlocked(&engine, k);
+        must_get_none(&engine, k, 28);
+        // Currently we cannot avoid this.
```

Comment:
Indeed, it isn't reasonable a committed transaction can re-acquire the same lock successfully. To avoid it, we need to call `get_txn_commit_info` in `acquire_pessimistic_lock`. It brings some overhead so I don't think we should fix it since it's impossible.

Reply:
Acquiring a pessimistic lock again after rolled back doesn't break data consistency.
I think this is acceptable.